privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ui/RequestDataViewStateMapper.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.privacy.dashboard.impl.ui
+
+import androidx.core.net.toUri
+import com.duckduckgo.app.global.UriString
+import com.duckduckgo.app.global.extractDomain
+import com.duckduckgo.app.global.model.Site
+import com.duckduckgo.app.global.withScheme
+import com.duckduckgo.app.trackerdetection.model.Entity
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus.AD_ALLOWED
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus.ALLOWED
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus.BLOCKED
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus.SAME_ENTITY_ALLOWED
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus.SITE_BREAKAGE_ALLOWED
+import com.duckduckgo.app.trackerdetection.model.TrackerStatus.USER_ALLOWED
+import com.duckduckgo.app.trackerdetection.model.TrackingEvent
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.AllowedReasons
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.DetectedRequest
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.Reason
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.RequestDataViewState
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.State.Allowed
+import com.duckduckgo.privacy.dashboard.impl.ui.PrivacyDashboardHybridViewModel.State.Blocked
+import com.squareup.anvil.annotations.ContributesBinding
+import okhttp3.internal.publicsuffix.PublicSuffixDatabase
+import javax.inject.Inject
+
+interface RequestDataViewStateMapper {
+    fun mapFromSite(site: Site): RequestDataViewState
+}
+
+@ContributesBinding(AppScope::class)
+class AppSiteRequestDataViewStateMapper @Inject constructor() : RequestDataViewStateMapper {
+
+    private val publicSuffixDatabase = PublicSuffixDatabase()
+    private val allowedCategories = listOf(
```

Comment:
I have a generic comment. My gut feeling is that I've already seen a lot of this code in classes like `Site`, `SiteMonitor`, `TrackerDetector`, etc. I'm guessing all that code now lives here and we are not duplicating it? Just double checking.

Reply:
if we are talking about `allowedCategories`, this is specific to the privacy dashboard UI (that's why inside the ViewStateMapper), so not sure if similar logic is duplicated somewhere else. If so, I think is also ok, since the list of supported categories in the domain layer doesn't need to necessarily match the UI layer.

if you were referring to something else, please ask again :D